src/maxflow.rs

```diff
@@ -1 +1,224 @@
+#![allow(dead_code)]
+use crate::internal_queue::SimpleQueue;
+use crate::internal_type_traits::Integral;
+use std::cmp::min;
+use std::iter;
 
+impl<Cap> MfGraph<Cap>
+where
+    Cap: Integral,
```

Comment:
Actually the algorithm doesn't rely on the integrality.... Though it does rely on `Ord`. Not sure what we should do.

Reply:
One possible way is to define a trait dedicated for max-flow capacities, which is my previous implementation.  Refer to [this commit](https://github.com/TonalidadeHidrica/ac-library-rs/commit/c05ac18d2bd8b77f62edba445195d3b66e076690) to see the differences.  However this is a bit like a repetition of `Integral` trait in `internal_type_traits` module, that's why I simply reused it.
At least the original library guarantees only `int` and `ll`, I guess this is already sufficient.

By the way, can this implementation be applied to floating-point capacity?  I mean, don't we need some additional treatment of precision errors?

Reply:
Good question!
I believe so long as min(x, y) in {x, y} and x-x = 0, the algorithm work fine-ish. e.g. for floating point numbers without NaN / inf, polynomial, etc.
I said fine-ish since although the time complexity doesn't get worse, there are errors that are unavoidable without using multi-precision floating point number. Guaranteeing the accuracy of the returned flow value and flows on edges would be hard. Farthemore since cut is discrete, the min-cut verticis can be different from the correct one although the cut value should be similar.

In competitive programming, I think I have seen some problems that uses polynomial flows (or big-M instead) and floating point numbers (came from binary/ternary search, can be transformed to a search on rational numbers though require a lot more coding).
Though maybe it will be very rare in AtCoder at least, since ACL doesn't support them.